tensorflow/compiler/mlir/tensorflow/tests/region-control-flow-to-functional.mlir

```diff
@@ -168,6 +168,50 @@ func @testIf2Result(%arg0: tensor<i1>, %arg1: tensor<2xf32>) -> tensor<2xf32> {
 
 // -----
 
+// Do not skip incompatible cast for trivial transform.
```

Comment:
Not sure if this test is actually testing the changed code. We look through casts inside the IfRegion body, but the cast here is outside.

Reply:
Thanks for review! This testcase is exactly the one in the original issue. MatchCallArgs does not limit to ops inside the region, that is, it keeps tracking back the definingOp until it is not a cast op.

Reply:
Updated comment: Two testcases for IfOp are trivial transforms but one with extern incompatible cast and the other has non-extern incompatible cast.